UnicornRender/video/include/unicorn/video/Material.hpp

```diff
@@ -87,10 +87,54 @@ class Material
     /** @brief Returns color */
     UNICORN_EXPORT glm::vec3 GetColor() const;
 
+    /**
+     * @brief Sets texture cordinates
+     *
+     * @note Can't set texture coordinates without texture.
+     *
+     * @param x texture offset by x in pixels
+     * @param y texture offset by y in pixels
+     * @param width offset width in pixels
+     * @param height offset height in pixels
+     */
+    UNICORN_EXPORT void SetTextureCoordinates(int32_t x, int32_t y, int32_t width, int32_t height);
+
+    /**
+     * @brief Sets normalized texture cordinates
+     *
+     * @note Can't set texture coordinates without texture.
+     *
+     * @param x texture offset by x from 0.0 to 1.0
+     * @param y texture offset by y from 0.0 to 1.0
+     * @param width offset width from 0.0 to 1.0
+     * @param height offset height from 0.0 to 1.0
+    */
+    UNICORN_EXPORT void SetNormalizedTextureCoordinates(float x, float y, float width, float height);
+
+    /** @brief Drops previous set texture coordinates */
+    UNICORN_EXPORT void SetDefaultTextureCoodinates();
+
+    /**
+     * @brief Returns texture coordinates
+     *
+     * @note will generate warning if texture wasnt set
+     *
+     * @return texture coordinates
+    */
+    UNICORN_EXPORT glm::vec4 GetTextureCoordinates() const;
+
+    /**
+    * @brief Returns normalized texture coordinates
+    *
+    * @return normalized texture coordinates
+    */
+    UNICORN_EXPORT glm::vec4 GetNormalizedTextureCoordinates() const;
+
     /** @brief Signal for material update notification */
     wink::signal<wink::slot<void()>> DataUpdated;
 protected:
     glm::vec3 m_color;
+    glm::vec4 m_spriteCoordinates;
```

Comment:
I'm not sure if storing only normalized coordinates is the best approach.
Having explicit pixel coordinates helps with async textures where you don't yet know the actual size of the image but can already fill the sprite information. Also if you end up resizing your sprite atlas image (as in adding more frames/sprites) you'd want to use pixel coordinates, not the normalized ones. So in fact you only need normalized when you want to pass this information to the renderer, but not for the user.

I would think about storing both and having them work independently from texture (it doesn't make sense to lock the ability to set sprite coordinates when texture was not set - they should operate independently)
When texture is changed or when new coordinates are set (and texture is set) we can recalculate normalized ones and set them. I am not familiar with how sprite atlas images are typically used and if it's helpful to have normalized interface for setting texture coordinates. I feel like usually you'd use absolute values and normalized comes in handy _only_ when scaling your sprite resolutions which can also be mitigated with simply scaling the absolute values anyway... Not sure about normalized `set` interface, `get` makes sense for the render though.